run-tests.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+docker compose up -d --build --quiet-pull
+
+go test -v
```

Comment:
I think a set of `curl` commands here would suffice for a simple stack like this and would be universal to our various languages and wouldn't pull in more dependencies.

Reply:
My thinking here was that we may extend the tests in the future with other vault-specific tests as the repo grows. However, maybe it is a good idea to start with a simple shell. Let me try it in a different PR and compare.